fix(curriculum): correct legend placement in complaint form

pranshu-saraswat opened the pull request:
- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or Ona.



Closes #62222

majestic-owl448 reviewed (changes requested):
the solution need also to be updated. Also please double check that the stylesheet is still applied the same way and that the tests are still working

ilenia-magoni commented:
does it render correctly? do the tests work? please show how the example project renders, and that you run the tests and they all pass

pranshu-saraswat commented:
Hi @ilenia-magoni, thank you for running the tests and for this detailed feedback. I understand now that my HTML change also requires a corresponding fix in the JavaScript tests.

As this is my first contribution, I'm not yet familiar with the project's testing setup, and I don't have the experience to debug and fix the test files correctly.

Would you be able to provide some guidance on which tests need to be updated? I am happy to learn, but I understand if this is now too complex for a first issue.

Thank you again for your help.

pranshu-saraswat commented:
Hi @ilenia-magoni, I've just pushed the final update. This version corrects the HTML in both the seed and solution sections, and also fixes the tests in the hints section. I believe all the requested changes have now been addressed.

Thank you again for all your guidance!

majestic-owl448 reviewed (commented):
what about the css of the example project (which is also the solution)? can you show that it renders identically to the version that is currently live?

also can you show a screenshot of your terminal running the tests successfully?

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-customer-complaint-form/67279fe50237291f80eed8b8.md:
````diff
@@ -214,26 +214,26 @@ document.getElementById("complaint-description").value = "Less than 20 chars";
 ```
 
-When `#other-complaint` is checked and the value of `#complaint-description` is changed to a valid value, you should set its border color to `green`.
+When `#other-complaint` is checked and the value of `#complaint-description` is changed to a valid value, you should set the parent fieldset's border color to `green`.
````
this contradicts the user stories, if you need to make such a change for the tests to work, it can't be update here only

ilenia-magoni commented on package-lock.json:
package-lock.json should not be included in the PR

pranshu-saraswat committed:
fix(curriculum): correct legend placement in complaint form

pranshu-saraswat commented:
Hi @ilenia-magoni,

My apologies for all the back-and-forth. I have now pushed a clean commit that I believe addresses all of your feedback.

This update corrects the HTML structure, adds the full JavaScript solution, and leaves the original tests untouched. The pull request should now be ready for a final review.

Thank you for your incredible patience and guidance!
